File: md5-pop-reverse.js
var as = require('async');
var ReverseMd5 = require('reverse-md5')

var redis = require("redis");
var client = redis.createClient({});

var _len = 4;
var _queue = 'md5';

var reverseMd5 = ReverseMd5({
    lettersUpper: false,
    lettersLower: true,
    numbers: false,
    special: false,
    whitespace: false,
    maxLen: _len
})

as.forever(
    function (next) {
        client.BRPOP(_queue, 0, function (list, data) {

            if (data) {

                var hashed = data[1];
                console.log(hashed);

                console.log(
                    reverseMd5(hashed));
            }

            next();
        });
    },
    function (err) {
        console.error(err);
        client.quit();
        process.exit(0);
    }
);

File: md5-pop-reverse.py
import redis
import string

r = redis.Redis(host='localhost', port=6379, db=0)
con = 0

while True:
    try:
        data = r.brpop("md5")

        _a = data[1]

        print(_a)

        if (con % 50 == 0) and (len(_a) > 0):
            print(".", end="", flush=True)

    except Exception:
        print("error")
